www/class/centreonConfigCentreonBroker.php

```diff
@@ -835,6 +838,101 @@ public function updateConfig(int $id, array $values)
         return true;
     }
 
+    /**
+     * Find a broker config original value based on fieldIndex
+     *
+     * @param int $configId
+     * @param string $configKey
+     * @param int $fieldIndex
+     * @return string|null
+     */
+    private function findOriginalValueWithFieldIndex(int $configId, string $configKey, int $fieldIndex): ?string
+    {
+        $stmt = $this->db->prepare(
+            'SELECT config_value FROM cfg_centreonbroker_info
+            WHERE config_id = :configId
+            AND config_key = :configKey
+            AND fieldIndex = :fieldIndex'
+        );
+        $stmt->bindValue(':configId', $configId, \PDO::PARAM_INT);
+        $stmt->bindValue(':configKey', $configKey, \PDO::PARAM_STR);
+        $stmt->bindValue(':fieldIndex', $fieldIndex, \PDO::PARAM_STR);
+        $stmt->execute();
+
+        $row = $stmt->fetch();
+
+        return $row['config_value'] ?? null;
+    }
+
+    /**
+     * Retrieve lua original password if it hasn't change
+     *
+     * @param int $configId
+     * @param array<string,mixed> $values
+     */
+    private function revealLuaPasswords(int $configId, array &$values): void
+    {
+        foreach ($values['output'] as &$output) {
+            foreach (array_keys($output) as $key) {
+                if (
+                    preg_match('/^lua_parameter__value_(\\d+)$/', (string) $key, $matches)
+                    && $output["lua_parameter__value_{$matches[1]}"] === $this->passwordReplacementValue
```

Comment:
```suggestion
                    && $output["lua_parameter__value_{$matches[1]}"] === \CentreonAuth::PWS_OCCULTATION
```

Reply:
To be consistent with all the password_occultation we've created a constant 
https://github.com/centreon/centreon/blob/develop/www/class/centreonAuth.class.php#L45

Reply:
@jeremyjaouen Is it possible to use the contants of classes that do not come from the legacy?

Reply:
It works, but should I use it or should we keep the legacy separate ?